wandb/integration/keras/callbacks/metrics_logger.py

```diff
@@ -38,10 +38,17 @@ def __init__(self, log_batch_frequency: Optional[int] = None, *args, **kwargs):
             wandb.define_metric("batch/batch_step")
             # set all other batch/ metrics to use this step
             wandb.define_metric("batch/*", step_metric="batch/batch_step")
+    
+    def _get_lr(self, step):
+        try:
+            return self.model.optimizer.learning_rate.numpy().item()
+        except AttributeError:
+            return self.model.optimizer.learning_rate(step=step).numpy()
```

Comment:
Passing the step from `on_epoch_end` as `epoch` and from `on_batch_end` as `batch` will give the wrong result IMO (I am confident about this is a LR Scheduler is used).

The learning rate depends on the global iteration, which can be accessed via `self.optimizer.iterations`.

I think the change would be:

`self.model.optimizer.learning_rate(step=self.optimizer.iterations).numpy()`